tests/backup/backup_restore_basic_test.go

```diff
@@ -1442,3 +1442,188 @@ var _ = Describe("{BackupMultipleNsWithSameLabel}", func() {
 		CleanupCloudSettingsAndClusters(backupLocationMap, credName, cloudCredUID, ctx)
 	})
 })
+
+// AddMultipleNamespaceLabel add labels to multiple namespace, perform manual backup, schedule backup using namespace label selector and restore
+var _ = Describe("{AddMultipleNamespaceLabels}", func() {
+	var (
+		batchSize                  int
+		desiredNumLabels           int
+		backupLocationUID          string
+		cloudCredUID               string
+		clusterUid                 string
+		backupName                 string
+		credName                   string
+		backupLocationName         string
+		restoreName                string
+		periodicSchedulePolicyName string
+		periodicSchedulePolicyUid  string
+		scheduleName               string
+		firstScheduleBackupName    string
+		namespaceLabel             string
+		restoreNames               []string
+		bkpNamespaces              []string
+		cloudCredUidList           []string
+		contexts                   []*scheduler.Context
+		appContexts                []*scheduler.Context
+		scheduleRestoreMapping     map[string]string
+		fetchedLabelMap            map[string]string
+		labelMap                   map[string]string
+		err                        error
+	)
+	bkpNamespaces = make([]string, 0)
+	backupLocationMap := make(map[string]string)
+	scheduleRestoreMapping = make(map[string]string)
+	JustBeforeEach(func() {
+		StartTorpedoTest("AddMultipleNamespaceLabels", "Add multiple labels to namespaces, perform manual backup, schedule backup using namespace labels and restore", nil, 85583)
+		log.InfoD("Deploy applications")
+		batchSize = 10
+		desiredNumLabels = 1000
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+	It("Add multiple labels to namespaces, perform manual backup, schedule backup using namespace label and restore", func() {
+		providers := getProviders()
+		Step("Validate applications", func() {
+			log.InfoD("Validate applications")
+			ValidateApplications(contexts)
+		})
+		Step("Adding labels to namespaces in multiple of 10 until 1000", func() {
```

Comment:
Why do we have to add in batches?

Reply:
Test case says add in batch of 10 till 1000